.github/workflows/win.yml

```diff
@@ -15,7 +15,7 @@ jobs:
           make deps
       - name: Cross-compile Crystal
         run: |
-          LLVM_TARGETS=X86 bin/crystal build --cross-compile --target x86_64-pc-windows-msvc src/compiler/crystal.cr -Dwithout_playground -Dwithout_iconv -Dwithout_interpreter
+          LLVM_TARGETS=X86 bin/crystal build --cross-compile --target x86_64-pc-windows-msvc src/compiler/crystal.cr -Dwithout_playground -Dwithout_iconv -Dwithout_interpreter -Duse_pcre2
```

Comment:
This first-generation compiler is still a compiler, so shouldn't it use PCRE1? Or are we intentionally detecting potential PCRE2 errors here? Same for the linker command below

Reply:
The reason to use PCRE2 directly also for the compiler is to drop PCRE completely from the Windows workflow. I guess we could also keep it there, at least for building the compiler. Wouldn't be too hard, but still extra effort.
My rationale is that Windows is still an unsupported preview platform and thus breaking PCRE compatibility isn't that much of a problem. Of course it's not ideal to have different compiler behaviour on different platforms either, though 🤔

Reply:
I guess it's just build both libraries, make sure to build the compiler with PCRE and distribute only PCRE2.

Reply:
Then wouldn't it break all users who try to build the compiler with `make -fMakefile.win FLAGS=-Duse_pcre`? The only compelling reason not to distribute PCRE1 right now is when the mirror dies, otherwise let's just try to keep things consistent between Windows and other platforms

Reply:
But then if `-Duse_pcre` is added to `Makefile.win`'s build recipe for the compiler as I suggested, then not distributing PCRE1 means users _must_ supply `FLAGS=-Duse_pcre2`.

Reply:
Yes, either that or manually install libpcre.

Reply:
That's rather poor UX. In that case please continue to omit `-Duse_pcre` in `Makefile.win`.

Back to this change, I don't think we need `-Duse_pcre2` here as it is already the new default.

Reply:
Why? The normal packages as for using crystal to build your (or someone else's) software. Not for building the compiler. We're not shipping a copy of `libllvm` either - which you need for building the compiler even more than PCRE.

Reply:
The reason we don't ship LLVM is because the static libraries are too large and we haven't figured out DLL distribution, not that we shouldn't ship it.

Reply:
Windows packaging is another rabbit hole we shouldn't get too deep into with this.

So I was actually wrong, we're still building and shipping PCRE on windows.